public/app/features/variables/state/__tests__/fixtures.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  BaseVariableModel,
+  ConstantVariableModel,
+  CustomVariableModel,
+  LoadingState,
+  QueryVariableModel,
+  VariableHide,
+  VariableOption,
+  VariableRefresh,
+  VariableSort,
+} from '@grafana/data';
+
+export function createBaseVariableModel(): Omit<BaseVariableModel, 'type'> {
```

Comment:
I think that having the same general template to all of them is good, so isn’t it better to call `createBaseVariableModel` with a parameter `type: VariableType` as well and not remove the type property manually?

Reply:
I don't think it would, having a return type of `BaseVariableModel` and being called `createBaseVariableModel`. It just creates an easier to grasp consistency. 
Should I omit properties I want to be set from outside (`BaseVariableModel`), or should I demand them to be required (`VariableOption`) when creating a new object? 
Both approaches are found here so it allows for further differences in upcoming mocks.